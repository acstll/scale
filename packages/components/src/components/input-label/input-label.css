/**
 * @prop --input-label-font-size: Font size of the input-label
 * @prop --input-label-font-family: Font family of the input-label
 * @prop --input-label-padding-top: Padding top of the input-label
 * @prop --input-label-color: Color of the input-label
 */

.input-label {
  font-size: var(--input-label-font-size, 1rem);
  font-family: var(--input-label-font-family, unset);
  padding-bottom: var(--input-label-padding-bottom, 0.5rem);
  color: var(--input-label-color, unset);
}
