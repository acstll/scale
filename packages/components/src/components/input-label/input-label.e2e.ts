import { newE2EPage } from '@stencil/core/testing';

describe('t-input-label', () => {
  it('renders', async () => {
    const page = await newE2EPage();
    await page.setContent('<t-input-label/>');
    const element = await page.find('t-input-label');
    expect(element).toHaveClass('hydrated');
  });
});
