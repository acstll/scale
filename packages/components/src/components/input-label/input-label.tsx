import { Component, Prop, h } from '@stencil/core';
import { CssClassMap } from '../../utils/utils';
import classNames from 'classnames';

@Component({
  tag: 't-input-label',
  styleUrl: 'input-label.css',
  shadow: true,
})
export class InputLabel {
  /** (optional) Input text class */
  @Prop() public customClass?: string = '';
  /** (optional) Input text theme */
  @Prop() public theme?: string = '';

  public render() {
    return (
      <div class={this.getCssClassMap()}>
        <slot />
      </div>
    );
  }

  private getCssClassMap(): CssClassMap {
    return classNames(
      'input-label',
      this.customClass && this.customClass,
      this.theme && `input-label--theme-${this.theme}`
    );
  }
}
