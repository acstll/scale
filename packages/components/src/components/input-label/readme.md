# t-input-text



<!-- Auto Generated Below -->


## Properties

| Property      | Attribute      | Description                 | Type     | Default |
| ------------- | -------------- | --------------------------- | -------- | ------- |
| `customClass` | `custom-class` | (optional) Input text class | `string` | `''`    |
| `theme`       | `theme`        | (optional) Input text theme | `string` | `''`    |


## CSS Custom Properties

| Name                        | Description                    |
| --------------------------- | ------------------------------ |
| `--input-label-color`       | Color of the input-label       |
| `--input-label-font-family` | Font family of the input-label |
| `--input-label-font-size`   | Font size of the input-label   |
| `--input-label-padding-top` | Padding top of the input-label |


----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
