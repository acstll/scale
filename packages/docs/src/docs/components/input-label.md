---
title: Input label
description: Input label element
url: /docs/input-label
contributors:
  - eeegor
---

# Input label

Input label component

selector: `t-input-label`

## Default

```javascript
<t-input-label>Required</t-input-label>
```

<div class="demo-container">
  <t-input-label>Required</t-input-label>
</div>
