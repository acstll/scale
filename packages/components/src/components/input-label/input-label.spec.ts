import { newSpecPage } from '@stencil/core/testing';
import { InputLabel } from './input-label';

describe('Input Label', () => {
  let element;
  beforeEach(async () => {
    element = new InputLabel();
  });

  it('should match snapshot', async () => {
    const page = await newSpecPage({
      components: [InputLabel],
      html: `<t-input-label>Label</t-input-label>`,
    });
    expect(page.root).toMatchSnapshot();
  });

  it('should handle theme css class', () => {
    element.theme = 'default';
    expect(element.getCssClassMap()).toContain('input-label--theme-default');
  });

  it('should have a default css class', () => {
    expect(element.getCssClassMap()).toContain('input-label');
  });
});
