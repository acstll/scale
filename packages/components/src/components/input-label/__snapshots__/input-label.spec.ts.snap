// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Label should match snapshot 1`] = `
<t-input-label>
  <mock:shadow-root>
    <div class="input-label">
      <slot></slot>
    </div>
  </mock:shadow-root>
  Label
</t-input-label>
`;
